oat-sa/tao-core-ui-fe
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417), [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419), and [ADF-1418](https://oat-sa.atlassian.net/browse/ADF-1418)

Requires: 
 - [x] https://github.com/oat-sa/tao-calculator-fe/pull/33
 - [x] update the file [`package.json`](https://github.com/oat-sa/tao-core-ui-fe/pull/558/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R53-R58) once the companion PRs have been merged and released

### Summary

Update the devDependencies for supporting Node 18

Set Node 18 for the CI workflow.

BREAKING CHANGE
- replacement of Node-SASS by Dart-SASS: change in SCSS might be needed (compound selectors are no longer supported)
- upgrade of minimatch

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run test:cov`, `npm run coverage`
- test the bundling: `npm run build`
- test the linting: `npm run lint:src`, `npm run lint:test`


[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ADF-1418]: https://oat-sa.atlassian.net/browse/ADF-1418?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
